pkg/ca/config.go

```diff
@@ -0,0 +1,102 @@
+package ca
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"errors"
+	"time"
+
+	"github.com/darvaza-proxy/core"
+	"github.com/darvaza-proxy/darvaza/shared/x509utils"
+	"golang.org/x/exp/slices"
+)
+
+const (
+	Day  = 24 * time.Hour // Day is 24 hours
+	Year = 365 * Day      // Year is 365 days
+
+	// DefaultCertificateDuration is 90 days
+	DefaultCertificateDuration = 90 * Day
+	// DefaultCADuration is 10 years
+	DefaultCADuration = 10 * Year
+)
+
+// Config describes how the [CA] will operate
+type Config struct {
+	// ECDSA tells if certificates should be ECDSA instead of RSA
+	ECDSA bool
```

Comment:
I think that it would be better to have a `type` maybe called `algo`? field stating `ECDSA` as a `string`, just to be prepared for the future where we might have another algos too

Reply:
enum maybe @karasz? having a string opens a huge can of worms for error handling

Reply:
I can live with anything that is not bolted down